src/Kiota.Builder/CodeDOM/CodeBlock.cs

```diff
@@ -30,6 +30,14 @@ public override IEnumerable<CodeElement> GetChildElements(bool innerOnly = false
             return InnerChildElements.Values;
         return new CodeElement[] { StartBlock, EndBlock }.Union(InnerChildElements.Values);
     }
+    public void RenameChildElement(string oldName, string newName)
+    {
+        if (InnerChildElements.TryRemove(oldName, out var element))
```

Comment:
because the name in the dictionary will still be the old one, which then has side effects for methods that find child elements by name.
We probably have a bunch of weird effects when we rename things. @nikithauc noticed that in #1559. But since that PR is still going to take a while to get merged because of staffing changes, I'm pushing that method here so we can quickly start using it where needed.

Reply:
Makes sense. I hadn't considered that the children were in a dictionary. :-)